index.html

```diff
@@ -5156,17 +5198,89 @@ <h3>Overview</h3>
         and generating a JSON-LD document in expanded form for all
         <a>RDF literals</a>, <a>IRIs</a>
         and <a>blank node identifiers</a>.
-        If the <a data-link-for="JsonLdOptions">useNativeTypes</a> flag is set to <code>true</code>,
-        <a>RDF literals</a> with a
-        <a>datatype IRI</a>
-        that equals <code>xsd:integer</code> or <code>xsd:double</code> are converted
-        to a <a>JSON numbers</a> and <a>RDF literals</a>
-        with a <a>datatype IRI</a>
-        that equals <code>xsd:boolean</code> are converted to <code>true</code> or
-        <code>false</code> based on their
-        <a>lexical form</a>
-        as described in
-        <a class="sectionRef" href="#data-round-tripping"></a>.
+
+        If the <a data-link-for="JsonLdOptions">useNativeTypes</a> flag is set
+        to <code>true</code>, <a>RDF literals</a> are converted to JSON values
+        based on their <a>datatype IRI</a> and <a>lexical form</a>, as described
+        in <a class="sectionRef" href="#algorithm-16"></a>.
+      </p>
+
+      <table class="simple" id="literal-conversation-with-useNativeTypes>
+        <caption>Literal conversion when <a data-link-for="JsonLdOptions">useNativeTypes</a> is set to <code>true</code></caption>
+        <thead>
+          <tr>
+            <th>RDF datatype</th>
+            <th>RDF Literal</th>
+            <th>JSON</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <th><code>xsd:string</code></th>
+            <td>An RDF string literal</td>
+            <td>JSON string literal</td>
+          </tr>
+
+          <tr>
+            <th rowspan="3"><code>xsd:integer</code></th>
+            <td>Whole number in the <code>[-2<sup>53</sup> + 1, 2<sup>53</sup> + 1]</code> range, representable as per [[IEEE-754-2008]]</td>
+            <td><a>JSON Number</a></td>
+          </tr>
+
+          <tr>
+            <td>Whole number out of that range</td>
+            <td>JSON string literal</td>
```

Comment:
Actually, this needs to be serialized as a value object with `@type`. The integer `1e54` (`1000000000000000078291540404596243842305360299886116864`) is outside of that range, but the value object needs to include `"@type": "xsd:integer".